app/src/main/java/com/android/sample/ui/activity/CreateActivity.kt

```diff
@@ -553,7 +554,8 @@ fun CreateActivityScreen(
                       Toast.makeText(context, "You must select a location.", Toast.LENGTH_SHORT)
                           .show()
                     } else {
-                      attendees += profileViewModel.userState.value!!
+                      if (selectedOptionType == ActivityType.INDIVIDUAL.name)
```

Comment:
The !! are known to be unsafe, I suggest modifu^ying it to something like: 
profileViewModel.userState.value?.let { attendees += it } 
or attendees += profileViewModel.userState.value ?: emptyList()

Reply:
> I think you miss the fact that the '!!' are used to say : "I'm sure this is never null". Knowing that this screen is allowed to access iff we are logged in (i.e. the user is not null) then even if it's unsafe here we can use it correctly

Saying "this screen is allowed access iff we are logged in" is not enough. Because we never know, say in the future we decide to allow guests to create activities. I believe what you say could be correct if we have a check **within** the screen, that would display an error message such as "You are not logged in, you can not create an activity".
I guess one could argue that this check is done thanks to the `if (creator == "") ... else ...` but this check is, in my opinion, not explicit enough.

I agree with you @Yassyass181104 that, knowing how we implemented the things, this should not pose any issue. But I also agree with @hmaryam84 on the fact that it is a bad practice, and that such behaviour already caused us crashes, e.g. the crash of the app when we opened some activities in the last meeting with the coaches.

More generally, I'd advise you to never assume anything (e.g. "I'm sure this is never null"), and to always consider that the worst case could happen!

@hmaryam84 suggested : 
> profileViewModel.userState.value?.let { attendees += it }
or attendees += profileViewModel.userState.value ?: emptyList()
I'd approve these suggestions.
For you to implement them quicker, I'll do you a suggestion directly on GitHub, and you can accept it here directly.

Reply:
Didn't see this point of view, even thought norammly we can't be able to create activities being a guest but I totally agree with both of you for the fact that it's a bad practice